### PR TITLE
Lambda for sending a reminder email, checking for an ongoing relation…

### PR DESCRIPTION
**Pull request by rahilb**
- Create a lambda `hasOngoingRelationship` that will eventually call the new Supporter Experience api. Currently the `hasOngoingRelationship` is hard coded to return `true`.
- Create a lambda `sendInactivityEmail` that calls IDAPI to send a reminder to the user. 
- If `hasOngoingRelationship` returns true, update the user's last activity timestamp (using the `updateLastActivityTimestamp` lambda). 
- If `hasOngoingRelationship` returns false, call IDAPI to send a reminder email to the user (using the `sendInactivityEmail` lambda). 
`hasOngoingRelationship` is set to return true so this can be tested in CODE while the supex api is not ready. 

Additionally, this PR adds unit tests for the node lambdas:

```
[14:52:22][Step 1/1] [info]   The updateLastActivityTimestamp Lambda
[14:52:22][Step 1/1] [info]     ✓ should call the provided endpoint using the passed in args
[14:52:22][Step 1/1] [info]   1 passing (22ms)
```

### CODE REVIEW

File: node-lambdas/src/main/nodejs/hasOngoingRelationship.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ */
+
+function hasOngoingRelationship(userId) {
+    return new Promise((resolve, reject) => {
+        resolve({
+            hasOngoingRelationship: true,
+            userId: userId
+        });
+    });
+}
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/hasOngoingRelationship.js]: we can shorten this to 
```
function hasOngoingRelationship(userId) {
    return Promise.resolve({
        hasOngoingRelationship: true,
        userId: userId
    });
}
```

`Promise.resolve()` is a helper method that returns a promise that instantly resolves, for cases like this where you need a promise

File: node-lambdas/src/main/nodejs/hasOngoingRelationship.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+exports.handler = async (event, context) => {
+    return await hasOngoingRelationship(event.userId)
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/hasOngoingRelationship.js]: do we actually need `async/await` here? returning a promise from a normal function should be enough to let node know it has to wait for it. Worth trying imo

File: node-lambdas/src/main/nodejs/requestInactiveUserBatch.js [hunk 3]
```diff
@@ -8,20 +8,20 @@ const http = require('https');
 
 function requestInactiveUserBatch() {
     return new Promise((resolve, reject) => {
-        var options = {
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/requestInactiveUserBatch.js]: neat!

File: node-lambdas/src/main/nodejs/requestInactiveUserBatch.js [hunk 4]
```diff
@@ -8,20 +8,20 @@ const http = require('https');
...
+        const options = {
             host: process.env.IDAPI_ROOT,
             path: '/inactivity-stream/batch',
             headers: {
-                'X-GU-ID-Client-Access-Token': process.env.CLIENT_ACCESS_TOKEN,
+                'X-GU-ID-Client-Access-Token': 'Bearer ' + process.env.CLIENT_ACCESS_TOKEN,
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/requestInactiveUserBatch.js]: we can go prettier:
```javascript
`Bearer ${process.env.CLIENT_ACCESS_TOKEN}`
```
Or even (!!!)
`['Bearer', process.env.CLIENT_ACCESS_TOKEN].join(' ')`
(this one is just me hating manual string formatting and totally a personal preference up to you)

File: node-lambdas/src/main/nodejs/sendInactivityEmail.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    return new Promise((resolve, reject) => {
+        const options = {
+            host: process.env.IDAPI_ROOT,
+            path: '/inactivity-reminder/' + userId,
+            headers: {
+                'X-GU-ID-Client-Access-Token': 'Bearer ' + process.env.CLIENT_ACCESS_TOKEN,
+                'Content-Type': "application/json",
+            },
+            method: 'POST'
+        };
```
Comment (walaura) [inline node-lambdas/src/main/nodejs/sendInactivityEmail.js]: can we move this outside the promise scope?

File: node-lambdas/src/main/nodejs/requestInactiveUserBatch.js (hunk 4 above)
Comment (rahilb) [reply node-lambdas/src/main/nodejs/requestInactiveUserBatch.js]: Idioms are great, this is why I asked for your review! 😎

File: cloudformation/retention.yaml [hunk 6]
```diff
@@ -147,13 +183,38 @@ Resources:
...
+               "Choices": [
+                 {
+                   "Variable": "$.hasOngoingRelationship",
+                   "BooleanEquals": false,
+                   "Next": "SendReminderEmail"
+                 },
+                 {
+                   "Variable": "$.hasOngoingRelationship",
+                   "BooleanEquals": true,
+                   "Next": "UpdateLastActivityTimestamp"
```
Comment (jfsoul) [inline cloudformation/retention.yaml]: these choices are a bit up in the air until we know exactly what the supporter experience endpoint will do

File: node-lambdas/src/main/nodejs/sendInactivityEmail.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Call IDAPI to request a inactive user batch is published to SNS
```
Comment (jfsoul) [inline node-lambdas/src/main/nodejs/sendInactivityEmail.js]: Is this comment correct?
